### ISSUE TITLE
"Failed to get authorize url" when through Socks5 proxy at 4.7.2

### ISSUE THREAD

**Issue by chenshaoju** [h3poteto/whalebird-desktop#3859]
## Description
Sorry for my English.

When I set a Socks5 proxy in whalebird settings, I can't log in, returning an error message: "Failed to get authorize URL".

## How To Reproduce
1. Open whalebird
2. Set a socks5 proxy
3. try log in.

## Your Environment
 - OS: Manjaro Linux 22.0.0 / Kernel 6.0.8-1-MANJARO
 - Whalebird Version: 4.7.2
 - Instance: acg.mn

Screenshot:
Proxy:
![Screenshot_20221203_153016](https://user-images.githubusercontent.com/515763/205430360-f2885023-e6ae-4302-a56f-d2996a930bb9.png)

Login error:
![Screenshot_20221203_153043](https://user-images.githubusercontent.com/515763/205430366-547aca0a-4759-4ff5-8b39-335345684e75.png)

Logs:
![Screenshot_20221203_153123](https://user-images.githubusercontent.com/515763/205430369-806c3d5b-9f52-4a59-8f39-659e08bbecf4.png)

If any more information is needed, please feel free to ask.

Edit: The socks5 proxy is working well
![Screenshot_20221203_154023](https://user-images.githubusercontent.com/515763/205430508-95ec159f-3493-41b6-b704-e6db7bbc596c.png)

**Comment by h3poteto** [comment]
Does this only happen on acg.mn?
Did you try to log in to another mastodon server with proxy?

**Comment by chenshaoju** [comment]
Hi, I tested 3 sites, and they are all not working.

![Screenshot_20221203_233754](https://user-images.githubusercontent.com/515763/205449197-23c11ffa-26b7-4d2e-9815-c90a95145286.png)

![Screenshot_20221203_233833](https://user-images.githubusercontent.com/515763/205449199-3fc65f0d-8675-4df5-845a-62d7c69cd802.png)

![Screenshot_20221203_233910](https://user-images.githubusercontent.com/515763/205449201-a1d6647d-033f-4aa5-ab79-edc8fcdf5c69.png)

**Comment by h3poteto** [comment]
OK, thanks.

**Comment by h3poteto** [comment]
![Screenshot from 2022-12-11-22-26-37](https://user-images.githubusercontent.com/4631959/206906353-a9ce2001-b01f-423f-9b62-7f0dc88715a9.png)

I can reproduce it. I was using local tor proxy
![Screenshot from 2022-12-11-22-25-36](https://user-images.githubusercontent.com/4631959/206906381-cb263423-bd58-4d94-b2f4-a8fd73292866.png)

**Comment by h3poteto** [comment]
```
  22:28:11.342 › TypeError: No "host"
      at parseSocksProxy (/run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/socks-proxy-agent/dist/index.js:29:15)
      at new SocksProxyAgent (/run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/socks-proxy-agent/dist/index.js:115:29)
      at proxyAgent (/run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/megalodon/lib/src/proxy_config.js:87:30)
      at /run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/megalodon/lib/src/megalodon.js:109:68
      at step (/run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/megalodon/lib/src/megalodon.js:48:23)
      at Object.next (/run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/megalodon/lib/src/megalodon.js:29:53)
      at /run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/megalodon/lib/src/megalodon.js:23:71
      at new Promise (<anonymous>)
      at __awaiter (/run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/megalodon/lib/src/megalodon.js:19:12)
      at detector (/run/media/h3poteto/ssd1/dev/src/github.com/h3poteto/whalebird-desktop/node_modules/megalodon/lib/src/megalodon.js:101:12)
```

**Comment by h3poteto** [comment]
The cause is megalodon: https://github.com/h3poteto/megalodon/issues/1508

**Comment by chenshaoju** [comment]
This issue has been fixed in 4.7.4 or another version, Thank you!
